src/main/java/com/spotify/reaper/resources/CommonTools.java

```diff
@@ -252,6 +253,25 @@ public static RepairSchedule storeNewRepairSchedule(
                                    repairParallelism, intensity,
                                    DateTime.now());
     scheduleBuilder.owner(owner);
+    
+    Collection<RepairSchedule> repairSchedules = context.storage.getRepairSchedulesForClusterAndKeyspace(repairUnit.getClusterName(), repairUnit.getKeyspaceName());
+    for(RepairSchedule sched:repairSchedules){
+    	Optional<RepairUnit> repairUnitForSched = context.storage.getRepairUnit(sched.getRepairUnitId());
+    	if(repairUnitForSched.isPresent() && repairUnitForSched.get().getClusterName().equals(repairUnit.getClusterName()) && repairUnitForSched.get().getKeyspaceName().equals(repairUnit.getKeyspaceName())){
+    		if((repairUnitForSched.get().getColumnFamilies().size()==0 && repairUnit.getColumnFamilies().size()==0)
+    				|| repairUnitForSched.get().getColumnFamilies().size()==0 && repairUnit.getColumnFamilies().size()!=0
+    				|| repairUnitForSched.get().getColumnFamilies().size()!=0 && repairUnit.getColumnFamilies().size()==0
+    				|| Sets.intersection(repairUnit.getColumnFamilies(),repairUnitForSched.get().getColumnFamilies()).size() >0){
```

Comment:
non-blocking:
could be made easier to read: if these two if statement conditionals were made into methods with intuitive descriptive names.

eg (names could be better…)
```
if (isEqual(repairUnitForSched, repairUnit) {
	if (isRepairInProgress(repairUnitForSched, repairUnit)) {
    		String errMsg = String.format(
                        "A repair schedule already exists for cluster \"%s\",keyspace \"%s\", and column families: %s",
                        cluster.getName(), 
                        repairUnit.getKeyspaceName(),
                        Sets.intersection(repairUnit.getColumnFamilies(),repairUnitForSched.get().getColumnFamilies()));
    				LOG.error(errMsg);
					throw new ReaperException(errMsg);
        }
}

private static boolean isValid(Optional<RepairUnit> repairUnitForSched, RepairUnit repairUnit) {
    return repairUnitForSched.isPresent() 
            && repairUnitForSched.get().getClusterName().equals(repairUnit.getClusterName()) 
            && repairUnitForSched.get().getKeyspaceName().equals(repairUnit.getKeyspaceName());
}

private static boolean isRepairInProgress(Optional<RepairUnit> repairUnitForSched, RepairUnit repairUnit) {
    return (repairUnitForSched.get().getColumnFamilies().size()==0 && repairUnit.getColumnFamilies().size()==0)
            || repairUnitForSched.get().getColumnFamilies().size()==0 && repairUnit.getColumnFamilies().size()!=0
    	    || repairUnitForSched.get().getColumnFamilies().size()!=0 && repairUnit.getColumnFamilies().size()==0
    	    || Sets.intersection(repairUnit.getColumnFamilies(),repairUnitForSched.get().getColumnFamilies()).size() >0;
}
```